src/renderer/viz/expressions/Zoomrange.js

```diff
@@ -0,0 +1,62 @@
+import BaseExpression from './base';
+import { pow, blend, linear, zoom, sub } from '../expressions';
+import { implicitCast, checkType } from './utils';
+
+/**
+ * Define a list of interpolated zoom ranges based on an input breakpoint list. Useful in combination with ramp (see examples).
+ *
+ * @param {Number[]} zoomBreakpointList - list of zoom breakpoints with at least two elements
+ * @return {Number}
+ *
+ * @example <caption>Set the width to 1 at zoom levels < 7, set the width at 20 at zoom levels > 10, interpolate between 1 and 20 at zoom levels in the [7,10] range.</caption>
+ * const s = carto.expressions;
+ * const viz = new carto.Viz({
+ *   width: s.ramp(s.zoomrange([7, 10]), [1, 20])
+ * });
+ *
+ * @example <caption>Set the width to 1 at zoom levels < 7, set the width at 20 at zoom levels > 10, interpolate between 1 and 20 at zoom levels in the [7,10] range. (String)</caption>
+ * const viz = new carto.Viz(`
+ *   width: width: ramp(zoomrange([7, 10]), [1, 20])
+ * `);
+ *
+ * @memberof carto.expressions
+ * @name zoomrange
+ * @function
+ * @api
+ */
+export default class Zoomrange extends BaseExpression {
+    constructor (zoomBreakpointList) {
+        zoomBreakpointList = implicitCast(zoomBreakpointList);
+
+        super({});
+        this.zoomBreakpointList = zoomBreakpointList;
+        this.type = 'number';
+        this.inlineMaker = inline => inline._impostor;
+    }
+    eval () {
+        return this._impostor.eval();
+    }
+    _bindMetadata (metadata) {
+        checkType('zoomrange', 'zoomBreakpointList', 0, 'number-array', this.zoomBreakpointList);
+        if (this.zoomBreakpointList.elems.length < 2) {
+            setTimeout(() => {
```

Comment:
I guess we are doing so because we need to launch this error immediately, but I do not understand why.

Reply:
Actually, I checked it and it's not needed, my bad